### PR TITLE
Enable OpenTelemetry tests for Ruby

### PR DESCRIPTION
**Pull request by marcotc**
## Description

This PR implements the parametric system-tests gRPC server to back the Ruby Tracer OpenTelmetry integration.

All tests are passing for Ruby after this PR merges.

Depends on https://github.com/DataDog/dd-trace-rb/pull/3026 to be released for tests to go green.

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
When we hit the rate limit (HTTP 429), retry after the time specified in
the response. This does not count toward the number of retries based on
other criteria.

### CODE REVIEW

File: tests/parametric/test_otel_span_methods.py [hunk 1]
```diff
@@ -110,13 +107,21 @@ def test_otel_set_attributes_different_types(self, test_agent, test_library):
             assert root_span["meta"]["d_bool_val"] == "false"
             assert root_span["meta"]["array_val_int"] == "[10, 20]"
             assert root_span["meta"]["array_val_double"] == "[10.1, 20.2]"
-        elif root_span["meta"]["language"] == "dotnet":
+        elif root_span["meta"]["language"] in "dotnet":
```
Comment (cbeauchesne) [inline tests/parametric/test_otel_span_methods.py:110]: Typo ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: tests/parametric/test_otel_span_methods.py (hunk 1 above)
Comment (marcotc) [reply tests/parametric/test_otel_span_methods.py:110]: This was `root_span["meta"]["language"] in ["dotnet", "ruby"]` earlier in this PRs lifetime, I'll revert this line.

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
ruby fails in CI/parametric, is it related ?

**Review by cbeauchesne — DISMISSED** [review]
Hi @marcotc , it's still failing on activated test, or did I missed something ? 

 https://github.com/DataDog/system-tests/actions/runs/6937541248/job/18871827685?pr=1456#step:4:1024

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**